src/com/facebook/buck/go/testmaingen.go

```diff
@@ -0,0 +1,437 @@
+/*
+ * Copyright 2015-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Generates testmain.go files from a group of test files.
+//
+// Regular go tests (ones run with `go test`) don't actually define a main
+// package. Moreover, Go's reflection does not have the ability to inspect
+// packages (e.g. list functions). This script generates a main.go that
+// runs some set of tests passed in on the CLI. The code liberally borrows from
+// the `go test` implementation at https://github.com/golang/go/blob/master/src/cmd/go/test.go
```

Comment:
Unfortunately not - the file itself depends on almost all other files in that directory. Most of those aren't needed for the test main generation, but they live in the same file. Bazel did something very similar, but removed some features (coverage & test case matching). See https://github.com/bazelbuild/bazel/blob/master/tools/build_rules/go/tools/generate_test_main.go .

I can alternatively try to not copy that code and write something buck-specific.